server/api/auth.ts

```diff
@@ -0,0 +1,44 @@
+import crypto from "crypto";
+import express from "express";
+
+import { ApiSessionReq, ApiSessionRes } from "../../common/types";
+import { Sessions, Users } from "../models";
+import { attachDoubleSubmit, generateToken } from "../security";
+
+const authApi = express.Router();
+
+authApi.post("/session", async (req, res) => {
+  const params: ApiSessionReq = req.body;
+
+  if (!params.doubleSubmit || params.doubleSubmit !== req.cookies["doubleSubmit"]) {
```

Comment:
`doubleSubmit` is more about implementation details but why not use some generic name like x-csrf-token? I think it's more readable for other devs as well.

Reply:
Good point. Also moved it into a header from the JSON body, it would be cleaner to use it that way on other requests in the future.